src/color-scale/color-scale.js

```diff
@@ -21,21 +21,98 @@ const Self = class ColorScale extends ColorElement {
 
 	connectedCallback () {
 		super.connectedCallback?.();
+		this._el.swatches.addEventListener("input", this, {capture: true});
 		this._el.swatches.addEventListener("colorchange", this, {capture: true});
+		this._el.swatches.addEventListener("click", this, {capture: true});
 	}
 
 	disconnectedCallback () {
 		this.#swatches = [];
+		this._el.swatches.removeEventListener("input", this, {capture: true});
 		this._el.swatches.removeEventListener("colorchange", this, {capture: true});
+		this._el.swatches.removeEventListener("click", this, {capture: true});
 	}
 
 	handleEvent (event) {
+		if (event.type === "input" && (!this.editable?.name || !event.target.matches("input[slot=before]"))) {
+			return;
+		}
+
+		let colors = Object.entries(this.colors);
+		let swatchIndex = this.#swatches.indexOf(event.target.closest("color-swatch"));
+		if (swatchIndex > -1) {
+			if (event.type === "colorchange") {
+				colors[swatchIndex][1] = event.target.color;
+			}
+			else if (event.type === "input") {
+				// Color name changed
+				colors[swatchIndex][0] = event.target.value;
+			}
+			else if (event.type === "click" && event.target.matches("[part=remove-button]")) {
+				colors.splice(swatchIndex, 1);
+				this.colors = Object.fromEntries(colors);
+				this.render();
+				return;
+			}
+
+			this.colors = Object.fromEntries(colors);
+		}
+
 		this.dispatchEvent(new event.constructor(event.type, {...event}));
 	}
 
 	propChangedCallback ({name, prop, detail: change}) {
-		if (name === "computedColors") {
+		if (name === "computedColors" && !this.editable?.name && !this.editable?.color) {
 			// Re-render swatches
+			// Only if nothing is being edited, otherwise the input would be lost
+			// or, e.g., "red" would be converted to "rgb(100%, 0%, 0%)" right after the typing is done
+			this.render();
+		}
+
+		if (name === "editable") {
+			let addButton = this._el.addButton;
+			if (this.editable?.color) {
+				if (!addButton) {
+					addButton = this._el.addButton = Object.assign(document.createElement("button"), {
+						id: "add-button",
+						part: "add-button",
+						textContent: "Add color",
+					});
+
+					addButton.addEventListener("click", evt => {
```

Comment:
We should abstract the functionality into a method so that authors can trigger it programmatically (whether or not the `editable` class is there)